src/HotChocolate/Data/src/Data/Filters/Expressions/QueryableFilterProvider.cs

```diff
@@ -38,7 +38,10 @@ public override FieldMiddleware CreateExecutor<TEntityType>(NameString argumentN
 
                 // next we get the filter argument.
                 IInputField argument = context.Field.Arguments[argumentName];
-                IValueNode filter = context.ArgumentLiteral<IValueNode>(argumentName);
+                IValueNode filter = context.ContextData.ContainsKey("filterValueNode")
+                    ? context.ContextData["filterValueNode"] as IValueNode ??
+                        throw new Exception()
```

Comment:
We need a proper `GraphQLException` here. Also, the exception should be moved into the throw helper.

Reply:
hmm i thought i did not commit these changes there are a few things wrong with this 😂 we should also have a constant as an identifier. my bad